### PR TITLE
Remove filtering from vw pin sale

### PR DESCRIPTION
**Pull request by wrridgeway**
See #149 for details.

### CODE REVIEW

**Comment by wrridgeway** [comment]
## reporting.vw_pin_most_recent_sale
```
new <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select *
from \"ci_149-remove-filtering-from-vw-pin-sale_reporting\".vw_pin_most_recent_sale
"
)

old <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select * from reporting.vw_pin_most_recent_sale
  "
)

> summary(new$sale_price)
integer64
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max.      NA's 
    10001    150000    250000    895982    412500 850000000    738908 
> summary(old$sale_price)
integer64
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max.      NA's 
    10001    150000    250000    895982    412500 850000000    738908 
```

**Comment by wrridgeway** [comment]
## reporting.vw_ratio_stats
```
new <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select *
from \"ci_149-remove-filtering-from-vw-pin-sale_reporting\".vw_ratio_stats
"
)

old <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select * from reporting.vw_ratio_stats
  "
)

> summary(new$ratio)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  0.000   0.777   0.908   0.951   1.021  48.184   31181 
> summary(old$ratio)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  0.000   0.777   0.908   0.951   1.021  48.184   31181 
```

**Comment by wrridgeway** [comment]
```
new <- dbGetQuery(
conn = AWS_ATHENA_CONN_NOCTUA,
"select *
from \"ci_149-remove-filtering-from-vw-pin-sale_reporting\".vw_res_report_summary
"
)

old <- dbGetQuery(
conn = AWS_ATHENA_CONN_NOCTUA,
"select * from reporting.vw_res_report_summary
"
)

> summary(new$sale_median)
integer64
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  16000  189000  286900  354485  436000 6280000    3754 
> summary(old$sale_median)
integer64
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  14605  188501  286800  359491  440000 6300000    363
```

**Comment by wrridgeway** [comment]
```
new <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select *
from \"ci_149-remove-filtering-from-vw-pin-sale_reporting\".vw_town_sale_history
"
)

old <- dbGetQuery(
  conn = AWS_ATHENA_CONN_NOCTUA,
  "select * from reporting.vw_town_sale_history
"
)

> summary(new$sale_median)
integer64
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  19000  143344  230462  273812  345452 3350000 
> summary(old$sale_median)
integer64
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  19000  143707  230328  273776  345000 3350000
```

Differences are a product of switching outlier type. Outlier exclusion only applies to years 2014 and later.

**Comment by wrridgeway** [comment]
```
with old as (
select year, count(*) as old from default.vw_pin_sale
group by year
),
new as (
select year, count(*) as new
from "ci_149-remove-filtering-from-vw-pin-sale_default".vw_pin_sale
where NOT sale_filter_less_than_10k
and NOT sale_filter_same_sale_within_365
and NOT sale_filter_deed_type
group by year
)

select old.year, old.old, new.new, old.old - new.new as diff
from old
left join new on old.year = new.year
```

year | old | new | diff
-- | -- | -- | --
2000 | 102957 | 102957 | 0
2015 | 82148 | 82148 | 0
2001 | 105056 | 105056 | 0
1973 | 1 | 1 | 0
2016 | 97973 | 97973 | 0
2014 | 88741 | 88741 | 0
2005 | 132874 | 132874 | 0
2018 | 97157 | 97157 | 0
2011 | 57936 | 57936 | 0
2007 | 86767 | 86767 | 0
2019 | 92103 | 92103 | 0
2012 | 72987 | 72987 | 0
2008 | 61607 | 61607 | 0
2003 | 120105 | 120105 | 0
2004 | 129456 | 129456 | 0
2022 | 98239 | 98239 | 0
2010 | 58500 | 58500 | 0
2013 | 92508 | 92508 | 0
2009 | 59730 | 59730 | 0
2023 | 55364 | 55364 | 0
2006 | 110040 | 110040 | 0
2021 | 104000 | 104000 | 0
2002 | 114439 | 114439 | 0
1999 | 806 | 806 | 0
2017 | 95673 | 95673 | 0
2020 | 91225 | 91225 | 0

**Comment by wrridgeway** [comment]
sonic

File: aws-athena/views/default-vw_pin_sale.sql [hunk 1]
```diff
@@ -235,9 +232,15 @@ max_version_flag AS (
...
-        sf.sv_is_heuristic_outlier,
+        CASE WHEN sf.sv_is_outlier = 1 THEN TRUE
+            WHEN sf.sv_is_outlier = 0 THEN FALSE
+        END AS sv_is_outlier,
+        CASE WHEN sf.sv_is_ptax_outlier = 1 THEN TRUE
+            WHEN sf.sv_is_ptax_outlier = 0 THEN FALSE
+        END AS sv_is_ptax_outlier,
+        CASE WHEN sf.sv_is_heuristic_outlier = 1 THEN TRUE
+            WHEN sf.sv_is_heuristic_outlier = 0 THEN FALSE
+        END AS sv_is_heuristic_outlier,
```
Comment (dfsnow) [inline aws-athena/views/default-vw_pin_sale.sql:243]: @wagnerlmichael Just merged a PR to fix this in the underlying tables, so this should be redundant. Let's revert it to the previous hunk.

File: aws-athena/views/default-vw_pin_sale.sql [hunk 2]
```diff
@@ -286,45 +288,40 @@ SELECT
     unique_sales.num_parcels_sale,
     unique_sales.buyer_name,
     unique_sales.sale_type,
-    sale_filter.sale_filter_lower_limit,
-    sale_filter.sale_filter_upper_limit,
-    sale_filter.sale_filter_count,
+    unique_sales.sale_filter_same_sale_within_365,
+    unique_sales.sale_filter_less_than_10k,
+    unique_sales.sale_filter_deed_type,
+    COALESCE(sales_val.sv_is_outlier, FALSE) AS sale_filter_is_outlier,
```
Comment (dfsnow) [inline aws-athena/views/default-vw_pin_sale.sql:294]: **suggestion:** Add an inline comment clarifying the difference between this and `sv_is_outlier` + a justification for it. I'll add the same to the docs.

File: aws-s3/scripts-ccao-data-warehouse-us-east-1/ccao-commercial_valuation.R
Comment (dfsnow) [inline aws-s3/scripts-ccao-data-warehouse-us-east-1/ccao-commercial_valuation.R:1]: **issue:** Can we move these changes into a separate PR, merge that one first, then merge the sales view changes?

**Review by dfsnow — APPROVED** [review]
This looks good now @wrridgeway, thanks for the cleanup. @wagnerlmichael once this is merged let's be sure to update sales val
